ICSharpCode.Decompiler/IL/Transforms/NullCoalescingTransform.cs

```diff
@@ -83,6 +82,20 @@ bool TransformRefTypes(Block block, int pos, StatementTransformContext context)
 				ILInlining.InlineOneIfPossible(block, pos, InliningOptions.None, context);
 				return true;
 			}
+			// stloc obj(valueInst)
+			// if (comp(ldloc obj == ldnull)) {
+			//		throw(...)
+			// }
+			// =>
+			// stloc obj(if.notnull(valueInst, throw(...)))
```

Comment:
This transform runs the risk of creating a dead store (though maybe not in practice, since `obj` would have already been inlined earlier in that case).
I'd add a check that `obj` has more than one use (load/address-taking).